File: Android-viaPatron/viaPatron/app/src/main/java/com/example/viapatron2/core/models/MyViewModel.java
package com.example.viapatron2.core.models;

import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.MutableLiveData;
import android.arch.lifecycle.ViewModel;

public class MyViewModel extends ViewModel {

    private MutableLiveData<UserTripRequestSession> requestSession;

    public LiveData<UserTripRequestSession> getRequestSession() {

        if (requestSession == null) {
            requestSession = new MutableLiveData<UserTripRequestSession>();
        }
        return requestSession;
    }

    public void setRequestSession(UserTripRequestSession updatedRequestSession) {

        requestSession.setValue(updatedRequestSession);
    }
}

File: viaPorter-Yuheng/app/src/main/java/com/example/viaporter/core/models/MyViewModel.kt
package com.example.viaporter.core.models

import android.arch.lifecycle.LiveData
import android.arch.lifecycle.MutableLiveData
import android.arch.lifecycle.ViewModel

class MyViewModel : ViewModel() {

    private var userInfo: MutableLiveData<List<String>>? = null

    fun getUserInfo(): LiveData<List<String>> {
        return userInfo ?: MutableLiveData()
    }

    private fun loadUserInfo() {
        // Todo: Do an asynchronous operation to fetch users.
    }
}
